File: src/main/java/com/partymakers/shareparty/domain/usecases/party/impl/RemovePartyExpenseImpl.java
package com.partymakers.shareparty.domain.usecases.party.impl;

import com.partymakers.shareparty.domain.usecases.party.RemovePartyExpense;
import com.partymakers.shareparty.domain.usecases.party.port.PartyExpensesRepository;

import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public class RemovePartyExpenseImpl implements RemovePartyExpense {

    private final PartyExpensesRepository partyExpensesRepository;

    @Override
    public void removePartyExpense(long expenseId, long partyId) {
        //TODO: удалять из  таблички с расходами, но нужно-ли?
        partyExpensesRepository.deleteExpense(expenseId, partyId);
    }
}

File: src/main/kotlin/com/partymakers/shareparty/party/domain/usecase/RemovePartyExpenseImpl.kt
package com.partymakers.shareparty.party.domain.usecase

import com.partymakers.shareparty.party.domain.entity.Expense
import com.partymakers.shareparty.party.domain.repository.PartyRoomRepository
import com.partymakers.shareparty.party.domain.exception.NotFoundException
import org.springframework.stereotype.Service

@Service
class RemovePartyExpenseImpl(
    private val partyRoomRepository: PartyRoomRepository
) : RemovePartyExpenseUseCase {

    override fun removePartyExpense(partyId: Long, expense: Expense) {
        val partyRoom = partyRoomRepository.findById(partyId)
            .orElseThrow { NotFoundException("Party room with id: $partyId not found") }

        val updatedPartyRoom = partyRoom.copy(
            expenses = partyRoom.expenses.filterNot { it == expense }
        )
        partyRoomRepository.save(updatedPartyRoom)
    }
} 